src/isar/state_machine/states/monitor.py

```diff
@@ -30,37 +39,54 @@ def start(self):
 
     def stop(self):
         self.iteration_counter = 0
+        if self.mission_status_thread:
+            self.mission_status_thread.join()
```

Comment:
Could we risk the code hanging on this step if something is holding up the thread?

Reply:
The code will hang if something is holding up the thread. It is here to make sure we don't have any threads going when we leave the state and make sure everything is closed properly. Could possibly kill the thread in some way, but I think it is bad practice. There should be some sort of timeout in the request module to limit the time we hang here, so should be a limited time in the worst case